java/jdbc/src/main/java/io/vitess/jdbc/VitessStatement.java

```diff
@@ -707,6 +643,14 @@ protected void checkSQLNullOrEmpty(String sql) throws SQLException {
         return updateCounts;
     }
 
+    protected void checkAndBeginTransaction() throws SQLException {
+        if (!(this.vitessConnection.getAutoCommit() || this.vitessConnection.isInTransaction())) {
```

Comment:
is this right? can you explain if so?  seems strange that if `isInTransaction` we begin a transaction.